catalog/ui/catalog-ui-search/src/main/webapp/component/visualization/table/table-viz.view.js

```diff
@@ -80,5 +107,48 @@ module.exports = Marionette.LayoutView.extend({
         }), {
             replaceElement: true
         });
+    },
+    exportAll: function() {
+        let data = {
+            hiddenFields: [],
+            columnOrder: user.get('user').get('preferences').get('columnOrder'),
+            columnAliasMap: properties.attributeAliases,
+            metacards: store.get('content').getCurrentQuery().get('result').get('results').fullCollection
+        };
+        $.ajax({
+            type: "POST",
+            url: '/search/catalog/internal/transform/csv?_=' + Date.now(),
+            data: JSON.stringify(data),
+            contentType: 'application/json',
+            success: (data, status,  xhr) => {
+                var filename = getFilenameFromContentDisposition(xhr.getResponseHeader('Content-Disposition'));
+                if (filename === null) {
+                    return;
```

Comment:
So there was a much simpler solution lurking here. If I don't receive (or can't parse) the filename I just generate a new one for them. Trivial and no error reporting/user interaction needed.